### PR TITLE
Supported events

### PR DESCRIPTION
**Pull request by Efnilite**
### Description
Adds a method to specify the supported events by this SyntaxElement. 
If the current parser event is not supported by the SyntaxElement, prints an error.
This removes the need for these checks in `init`s.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Review by APickledWalrus — COMMENTED** [review]
not a _huge_ fan of this given that it ties events deeper into the api, but i'd like to see what others think.

**Comment by sovdeeth** [comment]
Yeah I'd rather it stay as is for now, since this should be context-aware in the future when someone cough cough gets around to context api

**Review by Moderocky — APPROVED** [review]
I think this looks ok but I want somebody more familiar with parser to approve this.

**Comment by UnderscoreTud** [comment]
> not a _huge_ fan of this given that it ties events deeper into the api, but i'd like to see what others think.

If this is a big enough concern we could move the method into its own interface, maybe called `EventRestrictedSyntax`? Or something along those lines. Just so it's not embedded into the SyntaxElement class

File: src/main/java/ch/njol/skript/lang/SkriptParser.java [hunk 1]
```diff
@@ -225,7 +220,23 @@ public boolean hasTag(String tag) {
 								}
 							}
 							T element = info.getElementClass().newInstance();
-							if (element.init(parseResult.exprs, patternIndex, getParser().getHasDelayBefore(), parseResult)) {
+
+							if (element instanceof EventRestrictedSyntax eventRestrictedSyntax) {
+								Class<? extends Event>[] supportedEvents = eventRestrictedSyntax.supportedEvents();
+								if (!getParser().isCurrentEvent(supportedEvents)) {
+									StringJoiner joiner = new StringJoiner(", ", "the ", "");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/lang/SkriptParser.java:227]: Does this properly handle multiple supported events? specifically, using or/and:
``x can only be used in the a, b, or c events``

**Review by sovdeeth — APPROVED** [review]

**Pull request merged**